// angular-safeguard/dist/Driver.js
import { convertFromJSON, serializeDataToString } from './helpers';
var LOCKER_TEST_KEY = 'LOCKER_TEST_KEY';
var Driver = (function () {
    function Driver(storage) {
        this.storage = storage;
    }
    Driver.prototype.set = function (key, data, config) {
        this.storage.setItem(key, serializeDataToString(data), config);
    };
    Driver.prototype.get = function (key) {
        return convertFromJSON(this.storage.getItem(key));
    };
    Driver.prototype.has = function (key) {
        var storage = this.storage;
        return storage.hasItem ? storage.hasItem(key) : this.storage.hasOwnProperty(key);
    };
    Driver.prototype.remove = function (key) {
        this.storage.removeItem(key);
    };
    Driver.prototype.clear = function () {
        this.storage.clear();
    };
    Driver.prototype.key = function (index) {
        if (index === void 0) { index = 0; }
        return this.storage.key(index);
    };
    Driver.prototype.isSupported = function () {
        try {
            this.storage.setItem(LOCKER_TEST_KEY, LOCKER_TEST_KEY);
            this.storage.getItem(LOCKER_TEST_KEY);
            this.storage.removeItem(LOCKER_TEST_KEY);
        }
        catch (e) {
            return false;
        }
        return true;
    };
    return Driver;
}());
export { Driver };
