// angular-safeguard/dist/LockerConfig.js
import { Inject, Injectable, InjectionToken } from '@angular/core';
import { isNil } from './helpers';
export var LOCKER_USER_CONFIG = new InjectionToken('LOCKER_USER_CONFIG');
var DEFAULT_SEPERATOR = ':';
var DEFAULT_NAMESPACE = '';
var DEFAULT_FALLBACK = "session" /* SESSION */;
export var USER_CONFIG_PROVIDER = {
    provide: LOCKER_USER_CONFIG,
    useValue: {
        namespaceSeparator: DEFAULT_SEPERATOR,
        driverFallback: DEFAULT_FALLBACK,
        driverNamespace: DEFAULT_NAMESPACE
    }
};
var configNotNil = function (config, prop) { return !config || isNil(config[prop]); };
var ɵ0 = configNotNil;
var LockerConfig = (function () {
    function LockerConfig(config) {
        this.driverNamespace = configNotNil(config, 'driverNamespace') ? DEFAULT_NAMESPACE : config.driverNamespace;
        this.driverFallback = configNotNil(config, 'driverFallback') ? DEFAULT_FALLBACK : config.driverFallback;
        this.namespaceSeparator = configNotNil(config, 'namespaceSeparator') ? DEFAULT_SEPERATOR : config.namespaceSeparator;
    }
    LockerConfig.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    LockerConfig.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [LOCKER_USER_CONFIG,] },] },
    ]; };
    return LockerConfig;
}());
export { LockerConfig };
export { ɵ0 };
