// angular-safeguard/__ivy_ngcc__/dist/CookieStorage.js
import { Cookie } from './Cookie';
var CookieStorage = (function () {
    function CookieStorage() {
    }
    CookieStorage.prototype.hasItem = function (key) {
        return !!Cookie.get(key);
    };
    CookieStorage.prototype.getItem = function (key) {
        return Cookie.get(key);
    };
    CookieStorage.prototype.setItem = function (key, value, config) {
        Cookie.set(key, value, config);
    };
    CookieStorage.prototype.removeItem = function (key) {
        Cookie.remove(key);
    };
    CookieStorage.prototype.clear = function () {
        Object.keys(Cookie.getAll())
            .forEach(function (key) { return Cookie.remove(key); });
    };
    CookieStorage.prototype.key = function (index) {
        var cookies = Object.keys(Cookie.getAll());
        return cookies[index];
    };
    Object.defineProperty(CookieStorage.prototype, "length", {
        get: function () {
            return Object.keys(Cookie.getAll()).length;
        },
        enumerable: true,
        configurable: true
    });
    return CookieStorage;
}());
export { CookieStorage };
