// angular-safeguard/__ivy_ngcc__/dist/DriverTypes.js
import { InjectionToken } from '@angular/core';
import { PollyfillDriver } from './PolyfillDriver';
import { Driver } from './Driver';
import { MemoryStorage } from './MemoryStorage';
import { CookieStorage } from './CookieStorage';
export var DRIVERS;
(function (DRIVERS) {
    DRIVERS["LOCAL"] = "local";
    DRIVERS["SESSION"] = "session";
    DRIVERS["MEMORY"] = "memory";
    DRIVERS["COOKIE"] = "cookie";
})(DRIVERS || (DRIVERS = {}));
/* tslint:disable variable-name */
export var LocalStorageDriver = {
    storage: new PollyfillDriver(localStorage),
    type: "local" /* LOCAL */
};
export var SessionStorageDriver = {
    storage: new PollyfillDriver(sessionStorage),
    type: "session" /* SESSION */
};
export var MemoryStorageDriver = {
    storage: new PollyfillDriver(new MemoryStorage()),
    type: "memory" /* MEMORY */
};
export var CookieStorageDriver = {
    storage: new Driver(new CookieStorage()),
    type: "cookie" /* COOKIE */
};
/* tslint:enable variable-name */
export var LOCKER_DRIVER_TYPES = new InjectionToken('LOCKER_DRIVER_TYPES');
export var DRIVER_TYPES_PROVIDERS = [{
        provide: LOCKER_DRIVER_TYPES,
        multi: true,
        useValue: LocalStorageDriver
    }, {
        provide: LOCKER_DRIVER_TYPES,
        multi: true,
        useValue: SessionStorageDriver
    }, {
        provide: LOCKER_DRIVER_TYPES,
        multi: true,
        useValue: MemoryStorageDriver
    }, {
        provide: LOCKER_DRIVER_TYPES,
        multi: true,
        useValue: CookieStorageDriver
    }];
