// angular-safeguard/__ivy_ngcc__/dist/MemoryStorage.js
var _cache = new Map();
var getKeys = function () { return Array.from(_cache.keys()); };
var ɵ0 = getKeys;
var MemoryStorage = (function () {
    function MemoryStorage() {
    }
    MemoryStorage.prototype.hasItem = function (key) {
        return _cache.has(key);
    };
    MemoryStorage.prototype.getItem = function (key) {
        return _cache.get(key);
    };
    MemoryStorage.prototype.setItem = function (key, value) {
        _cache.set(key, value);
    };
    MemoryStorage.prototype.removeItem = function (key) {
        _cache.delete(key);
    };
    MemoryStorage.prototype.clear = function () {
        _cache.clear();
    };
    MemoryStorage.prototype.key = function (index) {
        return getKeys()[index];
    };
    Object.defineProperty(MemoryStorage.prototype, "length", {
        get: function () {
            return getKeys().length;
        },
        enumerable: true,
        configurable: true
    });
    return MemoryStorage;
}());
export { MemoryStorage };
export { ɵ0 };
